fs_streams_http/index.js

```diff
@@ -45,87 +58,169 @@ async function readEventsFromCSV() {
 
 app.use(bodyParser.json());
 
-app.get("/events", async (req, res) => {
-  const location = req.query.location;
-  const events = await readEventsFromCSV();
-
-  if (location) {
-    return res.json(events.filter((event) => event.location === location));
-  }
+app.get("/events", async (req, res, next) => {
+  asyncLocalStorage.run(req, async () => {
+    const location = req.query.location;
+    logger.info("incoming request for events", {
+      timestamp: new Date(),
+      location,
+    });
+    // error
+    const error = new Error("fatal error");
+    error.statusCode = 301;
+    next(error);
+    ///
+    const events = await readEventsFromCSV();
+    if (location) {
+      return res.json(events.filter((event) => event.location === location));
+    }
+    res.json(events);
+  });
+});
 
-  res.json(events);
+app.get("/events/:eventId", async (req, res, next) => {
+  asyncLocalStorage.run(req, async () => {
+    try {
+      const eventId = req.params.eventId;
+      logger.info("incoming request for event", {
+        timestamp: new Date(),
+        eventId,
+      });
+      const events = await readEventsFromCSV();
+      const event = events.filter(
+        (event) => event.id.toString() === eventId
+      )[0];
+
+      if (!event) {
+        return res.sendStatus(404);
+      }
+
+      res.json(event);
+    } catch (e) {
+      next(e);
+    }
+  });
 });
 
-app.get("/events/:eventId", async (req, res) => {
-  const eventId = req.params.eventId;
-  const events = await readEventsFromCSV();
-  const event = events.filter((event) => event.id.toString() === eventId)[0];
+app.post("/events", async (req, res, next) => {
+  asyncLocalStorage.run(req, async () => {
```

Comment:
there should be single `asyncLocalStorage.run(` which can be done in app.use() section